Autopilot/Src/timeStampingPOGI.cpp

```diff
@@ -0,0 +1,57 @@
+
+#include "TimeStampingPOGI.hpp"
+#include "tim.h"
+#include "stm32f7xx_hal_tim.h"
+
+
+TimeStampingPOGI::TimeStampingPOGI(){
+
+    //initializing and starting timer
+    MX_TIM12_Init();
+    HAL_TIM_Base_Init(&htim12);
+    HAL_TIM_Base_Start(&htim12);
+
+}
+
+
+//just typical singleton stuff
+
+//Allocates and initializes the static data member pointer (not the object itself)
+TimeStampingPOGI* TimeStampingPOGI::s_Instance = nullptr;
+
+//creates the instance if it doesnt exist & returns the pointer to the instance
+TimeStampingPOGI* TimeStampingPOGI::GetInstance(){
+
+    if(s_Instance == nullptr){
+        s_Instance = new TimeStampingPOGI();
+    }
+
+    return s_Instance;
+}
+
+
+void TimeStampingPOGI::setGPSTime(GpsData_t* gpsData){
+
+    //check if its new and set the recentUTCTime to it
+    if (gpsData->timeIsNew){
+        __HAL_TIM_SET_COUNTER(&htim12, 0);
+        recentUTCTime = gpsData->utcTime;
+    }
```

Comment:
You still need to make a method to convert the GPS time to UTC time in seconds since midnight

I want that method to do something like this calculator: https://www.tools4noobs.com/online_tools/hh_mm_ss_to_seconds/

Reply:
oops yeah, not sure how but it completely slipped my mind